FHEM/14_SD_UT.pm

```diff
@@ -1716,167 +1727,167 @@ sub SD_UT_Attr(@) {
 	my $bitData;
 	my $hex_lengh = length(InternalVal($name, "lastMSG", "0"));
 
-	if ($cmd eq "set" && $attrName eq 'repeats' && $attrValue !~ m/^[1-9]$/i) {
-		return "$name: Unallowed value $attrValue for the attribute repetition (must be 1 - 9)!";
-	}
-
-	############ chance device models ############
-	if ($cmd eq "set" && $attrName eq "model" && $attrValue ne $oldmodel) {
-
-		if (InternalVal($name, "bitMSG", "no data") ne "no data") {
-			my $devicemodel;
-
-			### ERROR for Users
-			my $allowed_models;
-			foreach my $keys (keys %models) {	# read allowed_models with the same hex_lengh
-				$allowed_models.= $keys.", " if ($models{$keys}{hex_lengh} eq $hex_lengh);
-			}
+	if ($cmd eq 'set') {
+		if ($attrName eq 'repeats' && $attrValue !~ m/^[1-9]$/xms) {
```

Comment:
Ich finde diese Schreibweise lesbarer und "klarer" im sinne von, hier springen wir raus wenn ....

```
return "$name: Unallowed value $attrValue for the attribute repetition (must be 1 - 9)!"
		if ($attrName eq 'repeats' && $attrValue !~ m/^[1-9]$/xms) 

```

Reply:
Genau diese Schreibweise bemängelt perlcritic. Finde ich eigentlich auch logisch. Wir schreiben und lesen nun mal von oben nach unten und von rechts nach links.
Nur ein Beispiel aus dem Code:

> Postfix control "if" used at line 1229, column 48. See pages 93,94 of PBP.

Reply:
Hmm, Zeile 1229 passt jetzt aber irgendwie nicht zu dieser: 1731

In PBP steht: "Reserve postfix if for flow-of-control statements" auf Seite 94.
Dort wird dann erläutert dass man es für "next, last, redo, return, goto, die, croak, or
throw*" verwenden soll.

Reply:
Die Zeile 1731 wird halt nicht bemängelt, weil ich sie perlcritic-konform formuliert habe. Deshalb habe ich beispielhaft wahllos irgend eine aus vielen anderen heraus gegriffen.

Seit dem ich bei 10_SD_GT alle bemängelten ifs nach vorn genommen habe, mäkelt perlcritic dort nicht mehr rum und so dachte ich, machst du es hier jetzt gleich richtig und dann kommst du :-)

Reply:
Bei mir wird das nicht bemängelt wenn links vom if ein return steht. 

Wenn vor dem Return was anderes steht, dann mag das stimmen.

Reply:
> Bei mir wird das nicht bemängelt wenn links vom if ein return steht.
> Wenn vor dem Return was anderes steht, dann mag das stimmen.

Stimmt, ich hab da jetzt nochmal genauer hingesehen, Zeilen die mit return beginnen, werden nicht bemängelt.

Reply:
Gut :)

Selbige gillt auch für :
next, last, redo, return, goto, die, croak, or
throw*"